src/Standards/PSR12/Sniffs/ControlStructures/ControlStructureSpacingSniff.php

```diff
@@ -101,7 +101,15 @@ public function process(File $phpcsFile, $stackPtr)
             $error = 'The first expression of a multi-line control structure must be on the line after the opening parenthesis';
             $fix   = $phpcsFile->addFixableError($error, $next, 'FirstExpressionLine');
             if ($fix === true) {
+                $phpcsFile->fixer->beginChangeset();
+                if ($tokens[$next]['line'] > ($tokens[$parenOpener]['line'] + 1)) {
+                    for ($i = ($parenOpener + 1); $i < $next; $i++) {
+                        $phpcsFile->fixer->replaceToken($i, '');
+                    }
+                }
```

Comment:
The principle of this fix is good. The one thing I'm concerned about is that the fixer now introduces a _new_ error (incorrect indentation), which then has to be fixed in a next fixer round.

Could this logic be written in a way that _indentation_ is not removed ?

Reply:
Yes, it would be more efficient to only remove non-indentation whitespace. I initially thought this would be complicated, but upon review can see a sensible way of achieving this. I've updated the pull request accordingly.